fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientReadPlatformServiceImpl.java

```diff
@@ -225,6 +228,7 @@ private String buildSqlStringFromClientCriteria(final SearchParameters searchPar
 
         String extraCriteria = "";
         if (sqlSearch != null) {
+        	SQLInjectionValidator.validateSQLInput(sqlSearch);
```

Comment:
I'd suggest to parse the params before we hit any depper layer ... so we can provide an early exit instead of relying to a depper level ...